loadtest/status.go

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) 2019-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+package loadtest
+
+import (
+	"encoding/json"
+	"errors"
+	"strings"
+	"time"
+)
+
+// State determines which state a loadtester is in.
+type State int
+
+// Different possible states of a loadtester.
+const (
+	StateStopped State = iota
+	StateStarting
+	StateRunning
+	StateStopping
+)
+
+// ErrUnknownState is returned when an unknown state variable is encoded/decoded.
+var ErrUnknownState = errors.New("unknown state")
```

Comment:
I would probably rename this to `ErrInvalidState`. It's a semantic detail really but there are no unknowns here. If it's not one of the declared ones then it's bad data.